tests/TestCase/Controller/Component/ImpersonateComponentTest.php

```diff
@@ -178,10 +188,13 @@ public function testLoginConfiged()
 
     /**
      * @return void
-     * @expectedException \Cake\Database\Exception
+     *
+     * @throws \Exception
      */
     public function testUnloadableUserModal()
     {
+        $this->expectException('\Cake\Database\Exception');
```

Comment:
A good practice is to move this right before the line that will throw the exception.

Reply:
Have moved all exception expectations to just before the login() method that throws them.